lib/mllp/sender.ex

```diff
@@ -343,6 +314,48 @@ defmodule MLLP.Sender do
     stop_connection(state, reason, "process terminated")
   end
 
+  @should_reconnect_errors [
+    :closed,
+    :enotconn,
+    :econnreset,
+    :enetreset,
+    :econnaborted,
+    :econnrefused
+  ]
+  defp handle_send_error(state, error) when error in @should_reconnect_errors do
```

Comment:
This function we should call in `attempt_connection` function also. May be we should rename it to handle_connection_error which could occur at send time or even when we are starting the process. e.g. when I do `MLLP.Sender.start_link("localhostA", 9000)` we get error `:nxdomain` and we keep trying to connect to it.

Reply:
I may be wrong, but that's the  job of `maintain_reconnect_time` is  it not? It's not optimal though I agree. But, the idea as I understand it was, ensure the connection timer is active, after 1 second (or less) the process will be sent  an info message which will attempt a reconnect. 

Are you thinking we should try to reconnect once and if it doesn't pan out bail?